components/bl-carousel-component/src/index.js

```diff
@@ -0,0 +1,124 @@
+import { useState, useEffect, useRef } from 'react';
+import { NextButton, PrevButton, List, CarouselIndicators } from './subcomponents';
+import { getAdjacentImages } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function Carousel({ component, eventHandlers }) {
+  const {
+    style,
+    display,
+    classList,
+    imagesData,
+    heightImage,
+    autoplayDelay,
+    animationDuration,
+    withControls,
+    withIndicators,
+    animationType
+  } = component;
+  const { onNextButton, onPrevButton, onMouseEnter, onMouseLeave } = eventHandlers;
+
+  const [imagesList, setImagesList] = useState(imagesData);
+  const [currentImg, setCurrentImg] = useState(0);
+  const [animation, setAnimation] = useState(null);
+  const [nextCurrentImage, setNextCurrentImage] = useState(0);
+  const [autoplay, setAutoplay] = useState(true);
+  const autoplayRef = useRef();
+
+  const { nextImg, prevImg } = getAdjacentImages(currentImg, imagesList.length);
+
+  useEffect(() => {
+    if (autoplayDelay && autoplay) {
+      autoplayRef.current = setTimeout(() => {
+        component.goToNextImage();
+      }, autoplayDelay);
+    } else if (!autoplay) {
+      clearTimeout(autoplayRef.current);
+    }
+  }, [currentImg, autoplayDelay, autoplay]);
+
+  component.goToNextImage = () => {
+    clearTimeout(autoplayRef.current);
+    setNextCurrentImage(nextImg);
+    setAnimation('next');
+
+    setTimeout(() => {
+      setAnimation(null);
+      setCurrentImg(nextImg);
+    }, animationDuration);
+  };
+
+  component.goToPrevImage = () => {
+    clearTimeout(autoplayRef.current);
+    setNextCurrentImage(prevImg);
+    setAnimation('prev');
+
+    setTimeout(() => {
+      setAnimation(null);
+      setCurrentImg(prevImg);
+    }, animationDuration);
+  };
+
+  component.goToImage = (index) => {
+    clearTimeout(autoplayRef.current);
+    setNextCurrentImage(index);
+
+    setAnimation( index < currentImg ? 'prev' : 'next')
```

Comment:
`setAnimation( index > currentImg ? 'next' : 'prev')`

Not critical, but positive conditions are easier to read, so try to avoid using negative if there is no any need.